### PR TITLE
On board openshift/knative-client to use OpenShift CI

### PR DESCRIPTION
**Pull request by navidshaikh**
openshift/knative-client is the client/CLI for knative, this PR
 on boards the client to use OpenShift CI and adds respective tests.

### CODE REVIEW

File: ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+tag_specification:
+  name: '4.0'
+  namespace: ocp
+promotion:
+  cluster: https://api.ci.openshift.org
+  namespace: openshift
+  name: release-next
+base_images:
+  base:
+    name: '4.0'
```
Comment (matzew) [inline ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml]: 4.1 ?

Comment (matzew) [reply ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml]: do you generate these?  check in the eventin / serving repos, we generate those files

File: ci-operator/jobs/openshift/knative-client/openshift-knative-client-release-next-presubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+presubmits:
```
Comment (matzew) [inline ci-operator/jobs/openshift/knative-client/openshift-knative-client-release-next-presubmits.yaml]: you did generate those, running the latest prow docker image?

**Comment by matzew** [comment]
are there no postsubmit hooks ?

File: ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml (hunk 1 above)
Comment (navidshaikh) [reply ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml]: I used the serving one, should bump to 4.1
https://github.com/openshift/knative-serving/blob/master/openshift/ci-operator/generate-ci-config.sh#L7

File: ci-operator/jobs/openshift/knative-client/openshift-knative-client-release-next-presubmits.yaml (hunk 2 above)
Comment (navidshaikh) [reply ci-operator/jobs/openshift/knative-client/openshift-knative-client-release-next-presubmits.yaml]: Yup, just rechecked there is `:latest` tag

**Comment by navidshaikh** [comment]
> are there no postsubmit hooks ?

Nope, probably we'll add one as needed later, want to get the setup in place ASAP.

File: ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml (hunk 1 above)
Comment (navidshaikh) [reply ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml]: fixed

Comment (stevekuznetsov) [reply ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml]: Still 4.0

File: ci-operator/config/openshift/knative-client/openshift-knative-client-release-v0.1.0.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+tag_specification:
+  name: '4.1'
+  namespace: ocp
+promotion:
+  cluster: https://api.ci.openshift.org
+  namespace: openshift
+  name: release-v0.1.0
+base_images:
+  base:
+    name: '4.0'
```
Comment (stevekuznetsov) [inline ci-operator/config/openshift/knative-client/openshift-knative-client-release-v0.1.0.yaml]: Your `tag_specification` here is 4.1 but your base is 4.0?

Comment (navidshaikh) [reply ci-operator/config/openshift/knative-client/openshift-knative-client-release-v0.1.0.yaml]: updated

Comment (navidshaikh) [reply ci-operator/config/openshift/knative-client/openshift-knative-client-release-v0.1.0.yaml]: thanks!

**Comment by navidshaikh** [comment]
[Updated](https://github.com/openshift/knative-client/commit/d3daf9747728adf55cf88605ced8e553b2287ee0) installing knative serving

**Comment by navidshaikh** [comment]
Updated to source knative/test-infra and added missing knative-serving namespace

/test pj-rehearse

**Comment by navidshaikh** [comment]
@stevekuznetsov : thanks for review, updated the PR, PTAL.

**Comment by petr-muller** [comment]
@navidshaikh I think you should also add the repo to be watched by Tide for automated merges ( around https://github.com/openshift/release/blob/master/cluster/ci/config/prow/config.yaml#L333).

**Pull request merged**